File: backend/test_backend.py
"""
Simple test script to verify the backend functionality
Run this after setting up the database and generating mock data
"""
import asyncio
import sys
import os
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from core.db import AsyncSessionLocal
from services.deadstock import get_dead_stock_analysis
from services.recommendations import generate_recommendation

async def test_backend():
    """Test core backend functionality"""
    print("🧪 Testing Backend Functionality...")
    
    async with AsyncSessionLocal() as db:
        try:
            # Test 1: Dead stock analysis
            print("\n1. Testing Dead Stock Analysis...")
            result = await get_dead_stock_analysis(db, days_since_last_sale=90, min_quantity=1)
            print(f"✅ Found {result['summary']['total_items']} dead stock items")
            print(f"   Total value: ${result['summary']['total_dead_stock_value']:,.2f}")
            
            # Test 2: Recommendations
            if result['items']:
                print("\n2. Testing Recommendations...")
                first_product_id = result['items'][0]['product_id']
                recommendation = await generate_recommendation(db, first_product_id)
                print(f"✅ Recommendation for product {first_product_id}: {recommendation['strategy']}")
            
            print("\n🎉 All tests passed! Backend is working correctly.")
            
        except Exception as e:
            print(f"❌ Test failed: {e}")
            return False
    
    return True

if __name__ == "__main__":
    success = asyncio.run(test_backend())

File: backend/scripts/testBackend.js
const axios = require('axios');

const API_BASE = 'http://localhost:8000';

class BackendTester {
  
  async runTests() {
    console.log('🧪 Testing Backend API Endpoints...\n');
    
    try {
      await this.testHealthEndpoint();
      await this.testDeadstockEndpoint();
      await this.testQueryEndpoint();
      await this.testRecommendationEndpoint();
      
      console.log('\n✅ All tests passed! Backend is working correctly.');
    } catch (error) {
      console.error('\n❌ Test failed:', error.message);
    }
  }
  
  async testHealthEndpoint() {
    console.log('1. Testing Health Endpoint...');
    const response = await axios.get(`${API_BASE}/api/health`);
    
    if (response.status === 200 && response.data.status === 'ok') {
      console.log('   ✅ Health check passed');
    } else {
      throw new Error('Health check failed');
    }
  }
  
  async testDeadstockEndpoint() {
    console.log('2. Testing Dead Stock Analysis...');
    const response = await axios.get(`${API_BASE}/api/deadstock`, {
      params: {
        daysSinceLastSale: 90,
        minQuantity: 1,
        limit: 10
      }
    });
    
    if (response.status === 200 && response.data.success) {
      const { summary, items } = response.data.data;
      console.log(`   ✅ Found ${summary.totalItems} dead stock items`);
      console.log(`   💰 Total value: $${summary.totalDeadStockValue.toFixed(2)}`);
    } else {
      throw new Error('Dead stock analysis failed');
    }
  }
  
  async testQueryEndpoint() {
    console.log('3. Testing Natural Language Query...');
    
    // Test predefined query (works without OpenAI)
    const response = await axios.post(`${API_BASE}/api/query`, {
      question: "Show me the most expensive products"
    });
    
    if (response.status === 200 && response.data.success) {
      const results = response.data.data.results;
      console.log(`   ✅ Query executed, returned ${results.length} results`);
    } else {
      throw new Error('Natural language query failed');
    }
  }
  
  async testRecommendationEndpoint() {
    console.log('4. Testing Recommendations...');
    
    // First get a product ID
    const productsResponse = await axios.get(`${API_BASE}/api/deadstock?limit=1`);
    
    if (productsResponse.data.data.items.length > 0) {
      const productId = productsResponse.data.data.items[0].productId;
      
      const response = await axios.post(`${API_BASE}/api/recommendations/${productId}`);
      
      if (response.status === 200 && response.data.success) {
        const recommendation = response.data.data;
        console.log(`   ✅ Recommendation generated: ${recommendation.strategy}`);
      } else {
        throw new Error('Recommendation generation failed');
      }
    } else {
      console.log('   ⚠️  No products available for recommendation test');
    }
  }
}

// Run tests if this file is executed directly
if (require.main === module) {
  const tester = new BackendTester();
  tester.runTests().catch(console.error);
}

module.exports = BackendTester;
